### PR TITLE
feat: Bazel support of "container_pull" dependency-type

### LINKED ISSUE #2905: Bazel: support updating docker digests/tags in container_pull
**What would you like Renovate to be able to do?**

I'd like Renovate to automatically update my docker dependencies declared in a bazel WORKSPACE file using [rules_docker's `container_pull`](https://github.com/bazelbuild/rules_docker#container_pull-1).

**Describe the solution you'd like**

Given

```
container_pull(
    name="<name>",
    registry="<docker_registry>",
    repository="<docker_repository>",
    digest="<image_digest>",
    tag="<image_tag>",
)
```

update `<image_tag>` and `<image_digest>` using the same behavior as described in https://renovatebot.com/docs/docker/

**Describe alternatives you've considered**

I'm not sure there are any.

**Additional context**

Note that `tag` is technically not required when `digest` is provided.

### PR DESCRIPTION
**Pull request by Dmitry9**
Automatically update docker dependencies declared in a bazel WORKSPACE file using [rules_docker's container_pull](https://github.com/bazelbuild/rules_docker#container_pull-1).

Given
```
container_pull(
    name="<name>",
    registry="<docker_registry>",
    repository="<docker_repository>",
    digest="<image_digest>",
    tag="<image_tag>",
)
```
update <image_tag> and <image_digest> using the same behavior as described in https://renovatebot.com/docs/docker/

Closes #2905

### CODE REVIEW

File: lib/manager/bazel/extract.js [hunk 1]
```diff
@@ -196,6 +216,21 @@ function extractPackageFile(content) {
...
+    } else if (
+      depType === 'container_pull' &&
+      currentValue &&
+      digest &&
+      repository &&
+      registry
+    ) {
+      dep.currentDigest = digest;
+      dep.currentValue = currentValue;
+      dep.purl = registry + '/' + repository;
```
Comment (rarkins) [inline lib/manager/bazel/extract.js]: purl is not necessary

File: lib/manager/bazel/extract.js [hunk 2]
```diff
@@ -196,6 +216,21 @@ function extractPackageFile(content) {
...
+      dep.depName = depName;
+      dep.repo = repository;
+      dep.datasource = 'docker';
+      dep.lookupName = dep.repo;
```
Comment (rarkins) [inline lib/manager/bazel/extract.js]: I think we can perhaps skip setting `dep.repo` if it is not used anywhere after. We can probably just use `depName` and `lookupName`.

**Review by rarkins — CHANGES_REQUESTED** [review]
Looks good. Please continue with testing

File: lib/manager/bazel/extract.js (hunk 1 above)
Comment (Dmitry9) [reply lib/manager/bazel/extract.js]: Done.

File: lib/manager/bazel/extract.js (hunk 2 above)
Comment (Dmitry9) [reply lib/manager/bazel/extract.js]: Done.

Comment (Dmitry9) [reply lib/manager/bazel/extract.js]:
```javascript
dep.datasource = 'docker';
```
cannot be skipped.

File: lib/manager/bazel/extract.js [hunk 3]
```diff
@@ -196,6 +216,19 @@ function extractPackageFile(content) {
       dep.lookupName = dep.repo;
```
Comment (rarkins) [inline lib/manager/bazel/extract.js]: I think this needs to be changed as dep.repo is not defined anymore

File: lib/manager/bazel/extract.js (hunk 2 above)
Comment (rarkins) [reply lib/manager/bazel/extract.js]: Agreed that dep.datasource needs to stay.

**Pull request merged**